src/LayerPlan.cpp

```diff
@@ -749,9 +754,30 @@ void LayerPlan::addWall(ConstPolygonRef wall, int start_idx, const GCodePathConf
             computeDistanceToBridgeStart((start_idx + point_idx - 1) % wall.size());
         }
 
-        addWallLine(p0, p1, non_bridge_config, bridge_config, flow, non_bridge_line_volume, speed_factor, distance_to_bridge_start);
+        // HACK ALERT
+        // the overlap compensation is not perfect, it can produce short non-flow reduced line segments within a sequence of flow reduced
+        // line segments and so to try and avoid printing the spurious fat line segments we require that their lengths are above a threshold
+
+        const coord_t max_spurious_fat_segment_length = 50; // microns
+
+        if (flow >= wall_min_flow && (first_line || !travel_required || vSize(p0 - p1) > max_spurious_fat_segment_length))
+        {
+            if (first_line || travel_required)
+            {
+                addTravel(p0, (first_line) ? always_retract : wall_min_flow_retract);
+                first_line = false;
+                travel_required = false;
+            }
+            addWallLine(p0, p1, non_bridge_config, bridge_config, flow, non_bridge_line_volume, speed_factor, distance_to_bridge_start);
+        }
+        else
+        {
+            travel_required = true;
```

Comment:
It seems illogical to set some `travel_required` variable here rather than just travelling immediately to `p1`.

Reply:
See reply above.